### PR TITLE
[Merged by Bors] - refactor(ContMDiff/Basic): multiplicativise, generalise and golf contMDiff_of_support and friends

### PR DESCRIPTION
**Pull request by grunweg**
- multiplicativise `contMDiff_of_support`, `contMDiffWithinAt_of_not_mem` and `contMDiffAt_of_not_mem` and use to_additive
- golf `extend_one` with the multiplicative version
- generalise these lemmas to manifolds with zero/one
- slight clean-up: remove unused variables and `open`s
- slight drive-by golfing of one proof
- deprecate `eventuallyEq_zero_nhds` in favor of `not_mem_tsupport_iff_eventuallyEq`

Addresses the post-merge review comments in #9669.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/ContMDiff/Basic.lean [hunk 1]
```diff
@@ -341,24 +326,27 @@ theorem smoothWithinAt_one [One M'] : SmoothWithinAt I I' (1 : M → M') s x :=
 end id
 
 /-- `f` is continuously differentiable if it is cont. differentiable at each `x ∈ tsupport f`. -/
-theorem contMDiff_of_support {f : M → F} (hf : ∀ x ∈ tsupport f, ContMDiffAt I 𝓘(𝕜, F) n f x) :
-    ContMDiff I 𝓘(𝕜, F) n f := by
+theorem contMDiff_of_support [Zero M'] {f : M → M'}
```
[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Geometry/Manifold/ContMDiff/Basic.lean]: Independently, I thought of the same (and found out about to_additive, luckily), just pushed..
One helper lemma needs a better home, and to_additive fails for some reason I don't understand yet.

**Comment by grunweg** [comment]
Just pushed an update addressing your changes. Now, `to_additive` fails in one more place, with the same error message.

**Comment by grunweg** [comment]
Sure, thanks for the offer!

**Comment by ADedecker** [comment]
I pushed a fix. The issue was that `to_additive` doesn't guess names, so it had no way to guess `eventuallyEq_one_nhds` was connected to `eventuallyEq_zero_nhds`. You can fix that using `@[to_additive existing foo]`, but I just used [not_mem_mulTSupport_iff_eventuallyEq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Support.html#not_mem_mulTSupport_iff_eventuallyEq) instead. Speaking of that, I think we should just remove `eventuallyEq_zero_nhds`, the statement with `tsupport` is better. On the other hand I like the shorter name, but I can't convince myself that it's worth keeping, so I'll remove it.

**Comment by ADedecker** [comment]
I've done everything I wanted, you can check that you're happy with it and ask another maintainer for a final review.

**Comment by grunweg** [comment]
@ADedecker Thanks for the fixes and further golf; looks good to me.
I just pushed a comment replacing the two remaining uses of the deprecated lemma, so the build should be fixed now.

**Comment by grunweg** [comment]
@maintainer Can somebody take a final look, please?

File: Mathlib/Geometry/Manifold/ContMDiff/Basic.lean [hunk 2]
```diff
@@ -341,24 +326,33 @@ theorem smoothWithinAt_one [One M'] : SmoothWithinAt I I' (1 : M → M') s x :=
 end id
 
 /-- `f` is continuously differentiable if it is cont. differentiable at each `x ∈ tsupport f`. -/
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Basic.lean:328]:
```suggestion
/-- `f` is continuously differentiable if it is cont. differentiable at 
each `x ∈ mulTSupport f`. -/
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by grunweg** [comment]
Thank you for the review, both of you!
bors r+

**Comment by ADedecker** [comment]
@grunweg FYI you can do `bors r+` again yourself

**Comment by ADedecker** [comment]
The build issue was due to [Function.left_id](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Function.html#Function.left_id) being deprecated recently in favor of [Function.id_comp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Function.html#Function.id_comp). I merged master and fixed it.

**Comment by grunweg** [comment]
Merged with master; the current bors batch *should* not conflict. (`MFDeriv` imports this file, hence its split should not affect this one.)
bors r+

**Comment by grunweg** [comment]
@ADedecker Thank you!

**Pull request closed without merging**